core/src/main/java/nu/yona/server/analysis/service/WeekActivityDto.java

```diff
@@ -121,24 +121,29 @@ public static WeekActivityDto createInstanceInactivity(UserAnonymizedDto userAno
 				includeDetail ? Optional.of(0) : Optional.empty(), new HashMap<>(),
 				IntervalActivity.hasPrevious(goal, startOfWeek, ChronoUnit.WEEKS),
 				IntervalActivity.hasNext(startOfWeek, ChronoUnit.WEEKS));
-		weekActivity.createRequiredInactivityDays(userAnonymized,
+		weekActivity.createRequiredInactivityDays(userAnonymized, earliestPossibleDate,
 				userAnonymized.getGoalsForActivityCategory(goal.getActivityCategory()), levelOfDetail, missingInactivities);
 		return weekActivity;
 	}
 
-	public void createRequiredInactivityDays(UserAnonymizedDto userAnonymized, Set<GoalDto> goals, LevelOfDetail levelOfDetail,
-			Set<IntervalInactivityDto> missingInactivities)
+	public void createRequiredInactivityDays(UserAnonymizedDto userAnonymized, LocalDate earliestPossibleDate, Set<GoalDto> goals,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This method has 21 lines, which is greater than the 20 lines authorized. Split it into smaller methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS138)